pytorch_lightning/trainer/data_loading.py

```diff
@@ -26,9 +26,14 @@
 
 
 def _has_len(dataloader: DataLoader) -> bool:
+    """ Checks if a given Dataloader has __len__ method implemented i.e. if
+    it is a finite dataloader or infinite dataloader """
     try:
         # try getting the length
-        _ = len(dataloader)
+        val = len(dataloader)
+        if val == 0:
```

Comment:
```suggestion
        if len(dataloader) == 0:
```
what about generator like dataloader with fix size but not accessible because of generator type